main.go

```diff
@@ -23,75 +14,96 @@ import (
 	"d7024e_group04/internal/node"
 	"d7024e_group04/internal/server"
 	"d7024e_group04/internal/store"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
 
 	"golang.org/x/sync/errgroup"
 )
 
 func main() {
+	rootCtx, cancelCtx := signal.NotifyContext(context.Background(), syscall.SIGTERM)
+	errGroup, errCtx := errgroup.WithContext(rootCtx)
+
+	me := createOwnContact()
+	routingTable := routingtable.NewRoutingTable(me)
+	memoryStore := store.NewMemoryStore()
+	client := client.NewClient()
+	kNetwork := &network.PublicNetwork{}
+	node := node.New(client, routingTable, memoryStore, kNetwork)
+
+	startServer(errGroup, errCtx, routingTable, memoryStore)
+	startAPI(errGroup, errCtx, node)
+	startCLI(errGroup, errCtx, cancelCtx, node)
+	startBootstrapping(errGroup, errCtx, node)
```

Comment:
If we are starting the API and CLI before the bootstrap, we should create an error response if the bootstrap have not yet finished.